code/modules/vending/_vending.dm

```diff
@@ -1396,26 +1425,26 @@
  * product_to_vend - the product record of the item we're trying to vend
  * price_to_use - price of the item we're trying to vend
  */
-/obj/machinery/vending/proc/proceed_payment(obj/item/card/id/paying_id_card, datum/data/vending_product/product_to_vend, price_to_use)
+/obj/machinery/vending/proc/proceed_payment(obj/item/card/id/paying_id_card, datum/data/vending_product/product_to_vend, price_to_use, mob/living/alive_user)
```

Comment:
`alive_user` is kinda a weird name for this, should give it a more accurate name for what the user is doing here, like I assume it's the mob paying

Also don't forget to update the doc comment

Also also putting this at the end is very weird, should put it after `paying_id_card` IMO (quick glance says there's only one use of this proc and one override so shouldn't be painful)